Fix unexpected attribute for PS4 Machine

MarkDordoy opened the pull request:
Inital push to handle issues when having PSv5 code in a script trying to be run on a device that only has psversion 4.

What I've done is split the script into three. One for determining the version and having the params passed. The second and third configure DSC, however one does for version 4 machines, one for version 5. 

@je-mandelbug FYI, please review. Runs without errors manually. We need to test config is actually on machine tomorrow.

Commit:
Inital push to handle issues when having PSv5 code in a script trying to be run on a device that only has psversion 4

MarkDordoy commented on ProvisionPullClient.ps1:
```diff
@@ -17,72 +28,28 @@
+Switch ($PSVersionTable.PSVersion.Major)
+{
+    5 {
+         Invoke-WebRequest https://raw.githubusercontent.com/justeat/PowerShellDSCUtils/master/Version5DSC.ps1 -OutFile "$PSScriptRoot\DSC\Version5DSC.ps1"
```
Hardcoding master here is an issue until these files are in master. Same for PS5 and PS4 file

ghost commented:
@MarkDordoy-JustEat does ProvisionPullClient work with get-help? From what I've read unless the help section is part of a function, it must be the first part of the script with the parameters coming after it. Other than that it looks how I was planning to do it 👍

Commit:
Moved help to top of file. As per Mark Lewis suggestion

Commit:
Removed the additional Dir level on set path

And pointed DSCLocal Config to the generated folder

Commit:
Fix path for calling secondary scripts

Commit:
Removed DSC folder from script path location

As the root is already in the DSC folder as that is where the script is run from.
